### PR TITLE
add instructions for running on FreeBSD 

### PR DESCRIPTION
**Pull request by Thefrank**
There is currently no official package for FreeBSD, I have added a section to "docs/workflow/requirements/freebsd-requirements.md" that covers how to run the SDK under FreeBSD.

This is a frequently asked question from those that are new to FreeBSD, those familiar with FreeBSD but not dotnet, and in https://github.com/dotnet/runtime/issues/14537

### CODE REVIEW

**Comment by am11** [comment]
Looks great, thanks!

Side note, (inspired by how CI does it to keep logs slightly tidier and comparable) I personally mount the volume with fixed path at root of container's FS:
```diff
- docker run --rm --volume $(pwd):$(pwd) --workdir $(pwd) --env ROOTFS_DIR=/crossrootfs/x64 -ti  $TAG ./build.sh -cross -os FreeBSD
+ docker run --rm --volume $(pwd):/runtime --env ROOTFS_DIR=/crossrootfs/x64 -ti "$TAG" \
+   /runtime/build.sh -cross -os FreeBSD
```

File: docs/workflow/requirements/freebsd-requirements.md [hunk 1]
```diff
@@ -65,3 +64,46 @@ sudo pkg install --yes libunwind icu libinotify lttng-ust krb5 cmake openssl nin
...
 It needs to be built once on supported platform or obtained via community resources.
 
+Running on FreeBSD
+-------------------
+Install the following packages:
+- icu
+- libunwind
+- lttng-ust (optional, debug support)
+- krb5
+- openssl (optional, SSL support)
```
Comment (wfurt) [inline docs/workflow/requirements/freebsd-requirements.md:74]: Isn't openssl part of base OS?
```
furt@d13:~ $ which openssl
/usr/bin/openssl

```

**Review by wfurt — APPROVED** [review]

File: docs/workflow/requirements/freebsd-requirements.md (hunk 1 above)
Comment (Thefrank) [reply docs/workflow/requirements/freebsd-requirements.md:74]: "yes but"
Yes! But that version is for the OS and things from ports should use other things from ports. 

Which can be confusing but generally...

If it ships with the base OS its for the base only (e.g., openssl, llvm/clang, libunwind) or for building the OS. If it comes from ports (something external from the base OS) , it should use programs from ports

Comment (wfurt) [reply docs/workflow/requirements/freebsd-requirements.md:74]: oh, well. I'm not going to argue.

**Comment by Thefrank** [comment]
@am11 I do something similar for that too as it keeps things cleaner for runs :)

I decided to leave it for now, if the syntax ends up being confusing or more people raise "build not working with docker" questions then I will change that to be a bit more clear.

**Comment by wfurt** [comment]
BTW it seems like we need python to build. I just bump to this while finally getting my build working again on fresh system

**Comment by wfurt** [comment]
Hmm. actually building is here: https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/freebsd-instructions.md

May need some updates as well. It may be better to point at the cross-building scripts instead of my old binary snapshot.

**Comment by Thefrank** [comment]
Those build instructions should be updated at some point. They also seem to duplicate the building instructions under https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/freebsd-requirements.md 

How should they be reworked/changed? Change docs/workflow/building/libraries/freebsd-instructions.md to point to docs/workflow/requirements/freebsd-requirements.md? This would be the easiest/laziest(?) way as the later includes building instructions using both Linux docker and native FreeBSD toolchain.

The docker instructions are fine for now but if I am updating the doc more I will update the container referenced and fix up the command as per @am11 's suggestion.

Native FreeBSD toolchain will need a dependency addition of `python3` which both net6 and net7 now need. Currently, using a community built net6 SDK as a bootstrap, it will build the current dotnet/runtime/HEAD with only 4 lines of changes to `/eng/Versions.props`. These will need to be added too.

I am fine with any of the above but also open to other ideas

**Comment by wfurt** [comment]
Yes, I'm on similar path. And https://lists.freebsd.org/pipermail/freebsd-current/2020-May/076043.html
Had to add it from ports.

I think freebsd-requirements.md should have instructions how to install/run runtime. The building document can reference it or duplicate the package for convenience. I like the one where you can select & paste commands to the shell and that makes it work.

**Pull request merged**